openedx/core/djangoapps/programs/utils.py

```diff
@@ -151,7 +184,12 @@ def progress(self, programs=None, count_only=True):
                 if self._is_course_complete(course):
                     completed.append(course)
                 elif self._is_course_in_progress(course):
-                    in_progress.append(course)
+                    course_in_progress = self.does_course_still_qualify_as_in_progress(now, enrolled_run_modes, course)
+                    course['expired'] = course_in_progress
+                    if course_in_progress:
```

Comment:
Using this bool to decide which group the course goes in makes sense. However, assigning the bool to the `expired` key seems wrong. If we keep things the way you have them now, if I'm enrolled in the course but it's "expired," it's still going to appear in the in progress section. We want it to move back to the not started group.

This seems more like what we want:
```py
is_in_progress = self._is_in_progress(...)
course['expired'] = not is_in_progress
if is_in_progress:
    in_progress.append(course)
else:
    not_started.append(course)
```